src/new/sendTransaction.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import EventEmitter from 'events';
+import { MIN_POLL } from '../constants';
+import transaction from '../transaction';
+import txApi from '../api/txApi';
+import txMiningApi from '../api/txMining';
+import { AddressError, OutputValueError, ConstantNotSet, MaximumNumberOutputsError, MaximumNumberInputsError } from '../errors';
+
+/**
+ * This is transaction mining class responsible for:
+ *
+ * - Submit a job to be mined;
+ * - Update mining time estimation from time to time;
+ * - Get back mining response;
+ * - Push tx to the network;
+ *
+ * It emits the following events:
+ * 'job-submitted': after job was submitted;
+ * 'estimation-updated': after getting the job status;
+ * 'job-done': after job is finished;
+ * 'send-success': after push tx succeeds;
+ * 'send-error': if an error happens;
+ **/
+class SendTransaction extends EventEmitter {
+  /*
+   * data {Object} Prepared tx data
+   */
+  constructor({
+    data=null,
+  } = {}) {
+    super();
+
+    this.data = data;
+    // Job estimation
+    this.estimation = null;
+    // Job ID
+    this.jobID = null;
+  }
+
+  /**
+   * Submit job to be mined, update object variables of jobID and estimation, and start method to get job status
+   * Emits 'job-submitted' after submit.
+   */
+  submitJob() {
+    // Get tx hex without parents and nonce
+    const txHex = transaction.getTxHexFromData(this.data);
+    // Send to be mined in tx mining API
+    txMiningApi.submitJob(txHex, false, true, (response) => {
+      this.estimation = response.expected_total_time;
+      this.jobID = response.job_id;
+      this.emit('job-submitted', {estimation: this.estimation, jobID: this.jobID});
+      this.handleJobStatus();
+    });
+  }
+
+  /**
+   * Schedule job status request
+   * If the job is done, emits 'job-done' event, complete and send the tx
+   * Otherwise, schedule again the job status request and emits 'estimation-updated' event.
+   */
+  handleJobStatus() {
+    // this.estimation and MIN_POLL are in seconds
+    const poll_time = Math.max(this.estimation / 2, MIN_POLL)*1000;
+
+    setTimeout(() => {
+      txMiningApi.getJobStatus(this.jobID, (response) => {
+        if (response.status === 'done') {
```

Comment:
Even though it should never happen, you may get a 404 if you take too long to check the status. Current tx-mining-server keeps the result for 5 minutes.

Reply:
I will handle this case. What do you think we should return? Unexpected error? Because it shouldn't happen

Reply:
I feel that unexpected error is a good one. It should definitely never happen. Maybe we should tell the user to check if the transaction has been sent looking into the history and try again if it hasn't.

Reply:
Yeah, it's better. Added this to the unexpected error